src/xpcc/architecture/platform/driver/can/generic/can_bit_timings.hpp

```diff
@@ -0,0 +1,101 @@
+// coding: utf-8
+/* Copyright (c) 2016, Roboterclub Aachen e.V.
+ * All Rights Reserved.
+ *
+ * The file is part of the xpcc library and is released under the 3-clause BSD
+ * license. See the file `LICENSE` for the full license governing this code.
+ */
+// ----------------------------------------------------------------------------
+
+#ifndef XPCC_COMMON_CAN_BIT_TIMINGS_HPP
+#define XPCC_COMMON_CAN_BIT_TIMINGS_HPP
+
+#include <xpcc/architecture/interface.hpp>
+
+#include "../../clock/generic/common_clock.hpp"
+
+namespace xpcc
+{
+
+/**
+ * CAN Bit Timing
+ *
+ * TODO: explain how this works
+ *
+ * This will try to archive a perfect bitrate match and the best possible
+ * sampling point.
+ *
+ * Example for CAN bit timing:
+ *   CLK on APB1 = 36 MHz
+ *   BaudRate = 125 kBPs = 1 / NominalBitTime
+ *   NominalBitTime = 8uS = tq + tBS1 + tBS2
+ * with:
+ *   tBS1 = tq * (TS1[3:0] + 1) = 12 * tq
+ *   tBS2 = tq * (TS2[2:0] + 1) = 5 * tq
+ *   tq = (BRP[9:0] + 1) * tPCLK
+ * where tq refers to the Time quantum
+ *   tPCLK = time period of the APB clock = 1 / 36 MHz
+ *
+ * STM32F1xx   tPCLK = 1 / 36 MHz
+ * STM32F20x   tPCLK = 1 / 30 MHz
+ * STM32F40x   tPCLK = 1 / 42 MHz
+ *
+ *
+ * @author	Kevin Laeufer
+ * @ingroup	can
+ */
+template<int32_t Clk, int32_t Bitrate, uint32_t DesiredSamplingPoint = 750000 /* 75% */>
```

Comment:
750.000 aka. 0.750000? How accurate does this sampling point need to be?
Why not just percent aka. 75?

`DesiredSamplingPoint` is not currently used here, is that because it would be used in `calcBS{1,2}()` instead of the LUT you have now?